roles/sap_install_media_detect/tasks/rename/add_zip_extension_loop_block.yml

```
@@ -0,0 +1,15 @@
+---
+
+- name: SAP Install Media Detect - Rename - Loop Block - Get info about target zip file '{{ line_item.file }}.zip'
+  ansible.builtin.stat:
+    path: "{{ line_item.dir }}/{{ line_item.file }}.zip"
+  register: __sap_install_media_detect_register_zip_file_stat
+
+- name: SAP Install Media Detect - Rename - Loop Block - Fail if '{{ line_item.file }}.zip' already exists
+  ansible.builtin.fail:
+    msg: "File {{ line_item.dir }}/{{ line_item.file }}.zip already exists!"
+  when: __sap_install_media_detect_register_zip_file_stat.stat.exists
+
+- name: SAP Install Media Detect - Rename - Loop Block - Rename '{{ line_item.file }}' to '{{ line_item.file }}.zip'
+  ansible.builtin.command: mv {{ line_item.dir }}/{{ line_item.file }} {{ line_item.dir }}/{{ line_item.file }}.zip
+  changed_when: true
```

Do we really want to fail if file already exists? Failing if file exists breaks idempotency in a way. 
May be we should not do anything? Or just save all files that were skipped to report them later at the end of the role execution? 

Parameter creates looks like a way to go to me. See example below

```suggestion
- name: SAP Install Media Detect - Rename - Loop Block - Move '{{ source }}' to '{{ destination }}'
  ansible.builtin.command: mv {{ source }}/{{ destination }}
  vars:
    source: "{{ line_item.dir }}/{{ line_item.file }}"
    destination: "{{ line_item.dir }}/{{ line_item.file }}.zip"
  args:
    creates: "{{ destination }}"
```

---

Thanks for these hints @kksat.
In the meantime, we have discussed the topic again among the other reviewers and we think we need a different approach: If there are two files with the same prefix but one with the correct suffix and one without suffix (e.g. `51050708_part4.rar` and `51050708_part4`, we behave as follows:
- If these files have the same checksum, we assume that the file without suffix is not required and just remove it unconditionally.
- If the files have a different checksum but are of the file type self-extracting rar (`exe`), `rar`, or `zip`, we use the parameter `sap_install_media_detect_rename_target_file_exists` to determine the behavior:
  - `skip`: Don't do anything - leave both files in place
  - `fail`: The role will simply fail
  - `overwrite`: The file with the correct suffix will be overwritten by the file without suffix. In other words, the file with suffix will have the content of the file without suffix.
- If the files have a different checksum but the file type of the file without extension is not of type self-extracting (`exe`), `rar`, or `zip`, the file will not be part of the list of files to be renamed, so these files will be left in place.